### PR TITLE
Feat/add tou

### PR DESCRIPTION
**Pull request by grignards**

### CODE REVIEW

File: src/branding/Config.js [hunk 1]
```diff
@@ -4,5 +4,7 @@ export const COLORS = {
 
 export const IS_SINGLE_SERVER = false
 
-export const SERVER_NAME = 'Community Tracim'
-export const SERVER_URL = 'community.tracim.fr'
+export const SERVER_NAME = 'Preprod Algoo Tracim'
+export const SERVER_URL = 'preprod-algoo.tracim.fr'
+// export const SERVER_NAME = 'Community Tracim'
+// export const SERVER_URL = 'community.tracim.fr'
```
Comment (gbonaspetti) [inline src/branding/Config.js:10]: I think we can leave Community here or just Preprod, but not both

File: src/modals/AcceptTermsOfUseModal.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { useTranslation } from 'react-i18next'
+import {
+  Linking,
+  Text,
+  TouchableOpacity as Button
+} from 'react-native'
+import { CustomModal } from './CustomModal.jsx'
+import { styles } from '../styles.js'
+import { modalStyles } from './modalStyles.js'
+import BouncyCheckbox from "react-native-bouncy-checkbox"
```
Comment (gbonaspetti) [inline src/modals/AcceptTermsOfUseModal.jsx:12]:
```suggestion
import BouncyCheckbox from 'react-native-bouncy-checkbox'
```

File: src/modals/AcceptTermsOfUseModal.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      <BouncyCheckbox
+        size={25}
+        fillColor="red"
+        unfillColor="#FFFFFF"
+        text="I've read and accept the terms of use"
+        iconStyle={{ borderColor: "red" }}
+        innerIconStyle={{ borderWidth: 2 }}
+        textStyle={{ fontFamily: "JosefinSans-Regular" }}
+        onPress={setAccepted}
```
Comment (gbonaspetti) [inline src/modals/AcceptTermsOfUseModal.jsx:47]: I just fixed the linting, but I have some questions:

- Why do we change the font?
- Wouldn't it be better to have these styles in a style sheet?
- Is it possible to add the translation to the text?

```suggestion
        size={25}
        fillColor='red'
        unfillColor='#FFFFFF'
        text="I've read and accept the terms of use"
        iconStyle={{ borderColor: 'red' }}
        innerIconStyle={{ borderWidth: 2 }}
        textStyle={{ fontFamily: 'JosefinSans-Regular' }}
        onPress={setAccepted}
```

File: src/authentificationHelper.js [hunk 4]
```diff
@@ -11,38 +17,156 @@ export const getCredentials = async (serverURL) => {
...
+
+/**
+ * Get the ID of the current user of a specific server using cache
+ * @param {String} serverURL URL of the server
+ * @returns The id of the connected user
+ */
+export const getCurrentUserID = async (serverURL) => {
+  let userId = -1
+
+  fetch(`https://${serverURL}/api/whoami`, {
```
Comment (gbonaspetti) [inline src/authentificationHelper.js:83]: All the other functions use try...(await)...catch, I wonder if this one should do the same to be more coherent. Idon't have an strong opinion about it though, you can leave like that if you like it more.

File: src/Tracim.jsx [hunk 5]
```diff
@@ -52,77 +47,48 @@ export const Tracim = () => {
...
+          {serverList.map(server => <Drawer.Screen
+				      component={ServerScreen}
+				      initialParams={{ server: server }}
+				      key={`drawer_${server.name}`}
+				      name={server.name}
+				      options={{
+					headerShown: false,
+					swipeEnabled: !IS_SINGLE_SERVER
+				      }}
+				    />)}
```
Comment (gbonaspetti) [inline src/Tracim.jsx:88]: Just a suggestion: 

```suggestion
          {serverList.map(server => (
            <Drawer.Screen
              component={ServerScreen}
              initialParams={{ server: server }}
              key={`drawer_${server.name}`}
              name={server.name}
              options={{
                headerShown: false,
                swipeEnabled: !IS_SINGLE_SERVER
              }}
            />
          ))}
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: src/modals/AcceptTermsOfUseModal.jsx (hunk 3 above)
Comment (grignards) [reply src/modals/AcceptTermsOfUseModal.jsx:47]: @Haskaris ?

File: src/modals/AcceptTermsOfUseModal.jsx (hunk 2 above)
Comment (grignards) [reply src/modals/AcceptTermsOfUseModal.jsx]: Done

File: src/modals/AcceptTermsOfUseModal.jsx (hunk 3 above)
Comment (grignards) [reply src/modals/AcceptTermsOfUseModal.jsx]: Seen with @Haskaris, I removed the custom style and font. Text is now passed through `t()`.

File: src/authentificationHelper.js (hunk 4 above)
Comment (grignards) [reply src/authentificationHelper.js]: This one is now unused, I removed it.

File: src/Tracim.jsx [hunk 6]
```diff
@@ -52,77 +47,50 @@ export const Tracim = () => {
...
-        />
-        {serverList.map(server => <Drawer.Screen
-          component={WebViewScreen}
-          initialParams={{ url: server.url }}
-          key={`drawer_${server.name}`}
-          name={server.name}
-          options={{
-            headerShown: false,
-            swipeEnabled: !IS_SINGLE_SERVER
+    serverList.length > 0 && (
```
Comment (gbonaspetti) [inline src/Tracim.jsx:61]: Why was this condition added? If the application is in multi server mode, it can have 0 elements in the list.

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: src/Tracim.jsx (hunk 6 above)
Comment (grignards) [reply src/Tracim.jsx:61]: To be able to initialize`initialRouteName` to the server in single server mode. But the test is indeed missing `!IS_SINGLE_SERVER`.

Comment (gbonaspetti) [reply src/Tracim.jsx:61]: Ok, I see the problem with `serverList[0].name`, I'm gonna change the condition

**Comment by grignards** [comment]
Le vendredi 4 novembre 2022, 14:30:56 CET gbonaspetti a écrit :
> @gbonaspetti commented on this pull request.
> 
> > -          component={HomeScreen}
> 
> -          name={t('Home')}
> -          options={{
> -            headerShown: false,
> -            swipeEnabled: !IS_SINGLE_SERVER
> -          }}
> -        />
> -        {serverList.map(server => <Drawer.Screen
> -          component={WebViewScreen}
> -          initialParams={{ url: server.url }}
> -          key={`drawer_${server.name}`}
> -          name={server.name}
> -          options={{
> -            headerShown: false,
> -            swipeEnabled: !IS_SINGLE_SERVER
> +    serverList.length > 0 && (
> 
> Ok, I see the problem with `serverList[0].name`, I'm gonna change the
> condition
Already done, just do a "pull --rebase"

**Review by gbonaspetti — APPROVED** [review]

**Pull request merged**